p2p_wallet/Scenes/Main/Swap/State/BusinessLogic/JupiterSwapBusinessLogic.swift

```diff
@@ -210,40 +211,127 @@ enum JupiterSwapBusinessLogic {
             guard state.status == .creatingSwapTransaction else {
                 return state
             }
-            
-            guard let route = state.route else {
+
+            guard var route = state.route else {
                 return state.error(.routeIsNotFound)
             }
 
             guard let account = state.account else {
                 return state.error(.createTransactionFailed)
             }
 
+            if available(.swapTransactionSimulationEnabled), state.routes.firstIndex(where: { $0.id == route.id }) == 0 {
+
+                var availableRoutes = state.routes
+                var swapTransaction: String?
+                var failedRoutes = [Route]()
+                for availableRoute in availableRoutes {
+                    // Try create and simulate transaction to see if it works correctly
+                    swapTransaction = try await createAndSimulateTransaction(for: availableRoute, account: account, services: services)
+
+                    if swapTransaction == nil {
+                        failedRoutes.append(availableRoute)
+                    } else {
+                        route = availableRoute
+                        // We found the best route and do not need to create transaction anymore
+                        break
+                    }
+                }
+
+                // Remove failed routes from the state
+                failedRoutes.forEach { failedRoute in
+                    availableRoutes.removeAll(where: { $0.id == failedRoute.id })
+                }
```

Comment:
I'm ok with the logic, but I think the removing algorithm is not so good. Since the routes have already been sorted better to worse, we may want to just iterate via its index and find the last index of routes that suite our need and omit the first indexes to that index using `array.dropFirst(N)`.
Store `failedRoutes` just to loop through them and remove it later is a little bit unnecessary.

Reply:
I agree. Fixed it